utils/build-script-impl

```diff
@@ -458,6 +458,13 @@ function set_build_options_for_host() {
     SWIFT_HOST_VARIANT_ARCH=$architecture
 
     case ${host} in
+        android-aarch64)
+            SWIFT_HOST_TRIPLE="aarch64-unknown-linux-android"
+            llvm_target_arch="AArch64"
+            swift_cmake_options=(
+                -DSWIFT_ANDROID_API_LEVEL:STRING="${ANDROID_API_LEVEL}"
+            )
+            ;;
```

Comment:
Yes, this is really meant for cross-compiling the host toolchain, but I need the host triple and API level when natively compiling the host toolchain on Android too, and since these flags are applied for native compilation of the host toolchain too, I went ahead and filled this out, following the instructions above to specify `llvm_target_arch` too.

I need the host triple because [I've always needed to set `LLVM_DEFAULT_TARGET_TRIPLE` on Android (scroll down to my `build-script-util` patch in that long comment to see how I used to hack it in before)](https://github.com/apple/swift/pull/23208#issuecomment-515830532), because LLVM assumes that it won't run on Android and it can't auto-detect the default triple for Android. I just found that this `SWIFT_HOST_TRIPLE` is ultimately used to set `LLVM_DEFAULT_TARGET_TRIPLE`, so I set it this way instead.

As for the API level, I needed it for native compilation so I stuck it here, but your suggestion is better so I will move that one alone, while keeping these other two here.

Reply:
OK. I didn't know that LLVM didn't autodetect Android triple. As a workaround, we can accept this, but I would not mind a comment explain why it is needed. I would also suggest that this might be something that LLVM should actually do, and upstreaming such patch might be interesting.

Reply:
No comment needed, as this triple is also definitely needed if cross-compiling the host toolchain, which will be the focus of my next pull. I have not looked into where LLVM auto-detects this, always manually specifying it instead, but you're right that it is worth adding.